pkg/updater/resultstore/resultstore.go

```diff
@@ -536,6 +538,59 @@ func processGroup(tg *configpb.TestGroup, group *invocationGroup) *updater.Infla
 	return col
 }
 
+// calculateMetrics calculates the numeric metrics (properties) for a targetResult
+func calculateMetrics(sar *singleActionResult) map[string]float64 {
+	properties := map[string][]string{}
+	testResultProperties(properties, sar.ActionProto.GetTestAction().GetTestSuite())
+	numerics := updater.Means(properties)
+
+	if dur := testResultDuration(sar.TargetProto); dur > 0 {
+		numerics[updater.ElapsedKey] = float64(dur) / 60000
+	}
+
+	return numerics
+
+}
+
+// testResultProperties recursively inserts all result and its children's properties into the map.
+func testResultProperties(properties map[string][]string, suite *resultstorepb.TestSuite) {
+
+	if suite == nil {
+		return
+	}
+
+	// add parent suite properties
+	for _, p := range suite.GetProperties() {
+		properties[p.GetKey()] = append(properties[p.GetKey()], p.GetValue())
+	}
+
+	// add test case properties
+	for _, test := range suite.GetTests() {
+		if tc := test.GetTestCase(); tc != nil {
+			for _, p := range tc.GetProperties() {
+				properties[p.GetKey()] = append(properties[p.GetKey()], p.GetValue())
+			}
+		} else {
+			testResultProperties(properties, test.GetTestSuite())
+		}
+	}
+}
+
+// testResultDuration calculates the overall duration of a target, converts
+// milliseconds to minutes and returns the duration.
+func testResultDuration(target *resultstorepb.Target) int64 {
```

Comment:
Comment from https://github.com/GoogleCloudPlatform/testgrid/pull/1253 still applies; I think this should return a time.Duration, or should specify milliseconds (not just in the comment, but in the function name or similar).

It also looks like this no longer returns the duration for the target's test results, just the target?

Reply:
Fixed!